react/utils/normalize.ts

```diff
@@ -84,14 +229,35 @@ const defaultSeller = { commertialOffer: { Price: 0, ListPrice: 0 } }
 const resizeImage = (url: string, imageSize: string | number) =>
   changeImageUrlSize(toHttps(url), imageSize)
 
+/**
+ * @description
+ * This method is responsible of constructing the product type for the product-context.
+ * The property `product.sku` is built by these specifications
+ * */
 export function mapCatalogProductToProductSummary(
-  product: any,
+  product: ProductTypes.Product,
+  preferredSKU: PreferenceType = 'FIRST_AVAILABLE',
   imageSize: string | number = 500
 ) {
   if (!product) return null
-  const normalizedProduct = { ...product }
+  const normalizedProduct: any = { ...product }
```

Comment:
Just a doubt, why do you need to type as any?

Reply:
I'm sending this as any because by typing the product in the function args, and referencing it again in the normalizeProduct, it would trigger the type check.

Reply:
Can you share where it will trigger the type check? It's breaking the typescript build? I can try to help you.a

Reply:
If you remove this type assignment, you'll see that on line 278 the property `sku` does not exist on `normalizedProduct` because when doing this: `const normalizedProduct = { ...product }` TS assigns the original type `ProductTypes.Product` to the new Object. 

You could intersect product and SKU, but you'll also need to type a new referenceId and the new images properties, this is way too much effort for so little gain...

Reply:
You're right, but have you ever tried to do something like that?

```suggestion
  const normalizedProduct: Product = { ...product }
```
Product comes from [product-summary-context](https://github.com/vtex-apps/product-summary-context/blob/master/react/ProductSummaryTypes.ts#L14)

If it doesn't work, you can try something like that:

```ts
const normalizedProduct: ProductTypes.Product & { sku?: SingleSKU} = { ...product }
```
[SingleSKU](https://github.com/vtex-apps/product-summary-context/blob/master/react/ProductSummaryTypes.ts#L52)

Reply:
We need to have everything typed, help others developers, and prevents us from miswriting code.

Reply:
![ex](https://user-images.githubusercontent.com/50715158/133332791-9f5192a9-6b67-4035-a505-2e9883227f46.png)

No because now this doesn't work... really where's the gain of typing something like this?

Reply:
The type is only used here, and nowhere else, and If we needed to arrange it to look like the original types, we would need to change the previous functions not assigned to this PR.

Reply:
This wasn't even typed before! I'm just getting the flak the because someone else didn't do it.

Reply:
@Git-the-Sanz I'm going to merge this PR now, but just to let you know for the future, this is definitely **not** how engineering at VTEX is treated. With this mindset of "This wasn't even typed before! I'm just getting the flak because someone else didn't do it. This "normalizedProduct" is not of my doing.". If we're making changes to a bit of code that you can improve, by typing, linting, simplifying... whatever it is, we do it. We're all working to make our products better, and we help each other looking over the quality of the code instead of blaming someone else for now having done something before.